benchmark/src/main/java/io/servicefabric/transport/protocol/ProtostuffMessageSerializerBenchmark.java

```diff
@@ -0,0 +1,109 @@
+package io.servicefabric.transport.protocol;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+
+import org.openjdk.jmh.annotations.*;
+
+import com.google.common.collect.ImmutableMap;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.servicefabric.cluster.gossip.Gossip;
+import io.servicefabric.cluster.gossip.GossipRequest;
+import io.servicefabric.transport.TransportHeaders;
+
+@Fork(2)
+@State(Scope.Thread)
+@Threads(4)
+@Warmup(iterations = 5)
+@Measurement(iterations = 5)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+public class ProtostuffMessageSerializerBenchmark {
```

Comment:
Also I think it is not representative compare since you put into message byte array data which has special processing at MessageSchema:

```
// Write data
		Object originalData = message.data();
		if (originalData != null) {
			if (originalData instanceof byte[]) {
				output.writeByteArray(DATA_PAYLOAD_FIELD_NUMBER, (byte[]) originalData, false);
			} else {
				Class<?> dataClass = originalData.getClass();
				Schema dataSchema = RuntimeSchema.getSchema(dataClass);
				try (RecyclableLinkedBuffer rlb = recyclableLinkedBuffer.get()) {
					byte[] array = ProtostuffIOUtil.toByteArray(originalData, dataSchema, rlb.buffer());
					output.writeByteArray(DATA_PAYLOAD_FIELD_NUMBER, array, false);
				}
				output.writeString(DATA_TYPE_FIELD_NUMBER, dataClass.getName(), false);
			}
		}
```

So it will wrote data as is without any additional manipulations (if statement). While in case of GossipRequest data it will go into "else" branch with all manipulations written there since Gossip Request is a Java Class. Obviously first approach much faster. But it will be more relevant to compare deserialization of Message with some Java POJO data (internally inside this POJO you can have byte array btw).